extensions/lifecycle/tasks/LifecycleObjectTask.js

```diff
@@ -0,0 +1,95 @@
+const async = require('async');
+const AWS = require('aws-sdk');
+
+const BackbeatTask = require('../../../lib/tasks/BackbeatTask');
+const AccountCredentials =
+    require('../../../lib/credentials/AccountCredentials');
+
+const attachReqUids =
+    require('../../replication/utils/attachReqUids');
+
+
+class LifecycleObjectTask extends BackbeatTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {QueueProcessor} qp - queue processor instance
+     */
+    constructor(qp) {
+        const qpState = qp.getStateVars();
+        super({
+            retryTimeoutS: qpState.repConfig.queueProcessor.retryTimeoutS,
+        });
+        Object.assign(this, qpState);
+
+        this.s3Credentials = null;
+        this.s3Client = null;
+    }
+
+    _setupClients(log, done) {
+        this.s3Credentials = new AccountCredentials(this.authConfig, log);
```

Comment:
`this.s3Credentials` is not used elsewhere in the class, so may be easier to just assign to a constant variable.

Reply:
Does this need to use `VaultClientCache`?

Reply:
Yes, it will need to make the lifecycle operation requests with an S3 client using the cached access/secret keys for the bucket owner. I've pushed the caching/key generation step in lifecycle producer now, so we can leverage the vaultclient cache here. Let me know if any changes are needed in the producer.